### ISSUE TITLE
Order of 'Select' data changed after replacing data source

### ISSUE THREAD

**Issue by Velociraptor45** [ant-design-blazor/ant-design-blazor#3750]
### Describe the bug

When I change the data source of a Select component, the order of its items is altered. The funny part is, the initially set data source does not have this problem. It also looks like this bug has been here before and came back (see #2439).

### Steps to reproduce (please include code)
We have two Selects here. The first one changes the data source of the second one.
```
<Select
    DataSource="_types"
    TItem="BaseType"
    TItemValue="int"
    LabelName="@nameof(BaseType.Name)"
    ValueName="@nameof(BaseType.Id)"
    DefaultValue="1"
    OnSelectedItemChanged="OnChange">
</Select>

<Select
    DataSource="GetDataSource(_type)"
    TItem="SubType"
    TItemValue="Guid"
    LabelName="@nameof(SubType.Name)"
    ValueName="@nameof(SubType.Id)"
    IgnoreItemChanges="false">
</Select>

@code {
    private int _type = 1;

    private readonly List<BaseType> _types = new()
    {
        new BaseType(1, "Type 1"),
        new BaseType(2, "Type 2")
    };

    private readonly List<SubType> _subTypeDs1 = new()
    {
        new(Guid.NewGuid(), "Item 1", 1),
        new(Guid.NewGuid(), "Item 2", 2),
        new(Guid.NewGuid(), "Item 3", 3),
        new(Guid.NewGuid(), "Item 4", 4),
        new(Guid.NewGuid(), "Item 5", 5),
        new(Guid.NewGuid(), "Item 6", 6),
        new(Guid.NewGuid(), "Item 7", 7),
        new(Guid.NewGuid(), "Item 8", 8),
        new(Guid.NewGuid(), "Item 9", 9)
    };

    private readonly List<SubType> _subTypeDs2 = new()
    {
        new(Guid.NewGuid(), "Item2 1", 1),
        new(Guid.NewGuid(), "Item2 2", 2),
        new(Guid.NewGuid(), "Item2 3", 3),
        new(Guid.NewGuid(), "Item2 4", 4),
        new(Guid.NewGuid(), "Item2 5", 5)
    };

    private void OnChange(BaseType newType)
    {
        _type = newType.Id;
        StateHasChanged();
    }

    private IEnumerable<SubType> GetDataSource(int type)
    {
        if (type == 1)
            return _subTypeDs1;

        return _subTypeDs2;
    }

    public record SubType(Guid Id, string Name, int SortingIndex);

    public record BaseType(int Id, string Name);
}
```

If you don't change anything and check out the second Select, the items are correctly ordered
![grafik](https://github.com/ant-design-blazor/ant-design-blazor/assets/30436688/003fc7b3-4da5-466a-a351-13cfd3797ea6)

Once you change the value in the first Select, the data source of the second Select gets updated and the order messed up
![grafik](https://github.com/ant-design-blazor/ant-design-blazor/assets/30436688/1db0509d-b18c-4046-8b2b-4d91be2d9dac)

And it's not data source specific. If you change back to "Type 1", the order keeps being wrong
![grafik](https://github.com/ant-design-blazor/ant-design-blazor/assets/30436688/d1288818-6941-43d6-bfd1-53a990cb0aa3)

I also tried to pass a `SortedSet<>` into it or change the records to classes, but nothing of that fixes it.

### Further technical details
- AntDesign Version: 0.18.1
- dotnet info:
```
.NET SDK:
 Version:           8.0.200
 Commit:            438cab6a9d
 Workload version:  8.0.200-manifests.e575128c

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
```

**Comment by NgMeiYann** [comment]
@Velociraptor45 , u need to add below code in the select component
` SortByLabel="SortDirection.Ascending"`

**Comment by Velociraptor45** [comment]
I don't want to sort it by label or Id, but by my own logic (here represented by the SortingIndex property). Thus, this sorting order won't fulfill my needs.

**Comment by GoldSucc** [comment]
@Velociraptor45 have you tried passing options as renderfragment? You can do so with <SelectOptions></SelectOptions>  fragment and for loop and each option should be <SelectOption>.

**Comment by Velociraptor45** [comment]
Hm, will try this over the weekend. Might work.
But this would be just a workaround. From my understanding this behavior is a bug and should be fixed.
But I will report back if it works.

**Comment by Velociraptor45** [comment]
Unfortunately, inserting the select options as a render fragment via `<SelectOption />` (see [here](https://antblazor.com/en-US/components/select#components-select-demo-label-in-value)) does not work at all for changing the data. The Select will only truncate the options list once you change to a shorter item list, but keep the original items in it. So, this is even worse than a wrong order. But thx for the suggestion.

**Comment by ElderJames** [comment]
Our contributior have fix this issue #3806 , will be released on this weekend. Thank you.
